dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/SdkFeatureTest.kt

```diff
@@ -549,6 +551,133 @@ internal class SdkFeatureTest {
 
     // endregion
 
+    // region Context Update Listener
+
+    @Test
+    fun `M register listeners W setContextUpdateListener()`(forge: Forge) {
+        // Given
+        val mockListeners = forge.aList(size = forge.anInt(min = 1, max = 10)) { mock<FeatureContextUpdateReceiver>() }
+
+        // When
+        val countDownLatch = CountDownLatch(mockListeners.size)
+        mockListeners.forEach {
+            Thread {
+                testedFeature.setContextUpdateListener(it)
+                countDownLatch.countDown()
+            }.start()
+        }
+
+        // Then
+        countDownLatch.await(5, TimeUnit.SECONDS)
```

Comment:
Alternatively this can be written without `CountDownLatch`:

```suggestion
        mockListeners.map {
            Thread {
                testedFeature.setContextUpdateListener(it)
                countDownLatch.countDown()
            }.apply { start() }
        }.forEach { it.join(5, TimeUnit.SECONDS) }

        // Then
```

but not a big deal